compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -7169,6 +7169,7 @@ public static class CommonAnalyzerData {
         boolean breakToParallelQueryEnv = false;
         int letCount = 0;
         boolean nonErrorLoggingCheck = false;
+        boolean withinAnnotationExpr = false;
```

Comment:
If we do not define `self` within the annotation scope, there is no need to define a special variable for it.